src/main/java/org/testeditor/fixture/web/angularjs/NgOneWebDriverFixture.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.fixture.web.angularjs;
+
+import java.util.concurrent.TimeUnit;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
+import org.testeditor.fixture.web.WebDriverFixture;
+
+import com.paulhammant.ngwebdriver.ByAngular;
+import com.paulhammant.ngwebdriver.NgWebDriver;
+
+public class NgOneWebDriverFixture extends WebDriverFixture {
+
+	private Logger logger = LogManager.getLogger(NgOneWebDriverFixture.class);
+	private NgWebDriver ngWebDriver;
+
+	@Override
+	protected WebElement getWebElement(String elementLocator) {
+		waitForAngularCompleteOperations();
+		if (elementLocator.startsWith("[model")) {
+			if (elementLocator.startsWith("[model(")) {
+				String substring = elementLocator.substring(elementLocator.indexOf("(") + 1);
```

Comment:
extracting this to a new method with a clear name, would make it much easier to read this code.